datastore/tests/fake_testdata.py

```diff
@@ -70,96 +72,40 @@ def fake_getter_run(
     yield getter_run
 
     # Update entities, metrics data etc after each GetterRun
-    # TODO: Update Publisher aggregate?
-    # TODO: Update Quality data? is that even possible or desirable for fake Grants?
     Latest.update()
     update_entities()
+    # Race conditions seem to happen when running tests if threads are enabled
```

Comment:
This function is doing quite a lot of heavy lifting... yield, atomic, context.. whilst also having data state changes. In the future I'd do as much as possible to try and simplify and make more explicit otherwise you're arguably going to need a unit test for `fake_getter_run`!